### ISSUE TITLE
react-easy-form breaks the dependency requirements

### ISSUE THREAD

**Issue by flosse** [cmsd2/rust-iron-react-webpack#2]
react-easy-form requires react `v0.14.0` :-\
Moreover the project seems to be dead.

**Comment by cmsd2** [comment]
damn. i'll take a look this week and see what I can do.

**Comment by cmsd2** [comment]
https://github.com/geowarin/react-easy-form/pull/5

It seems to upgrade cleanly to v15 which is good. The lack of activity could be a problem.

I find it hard to keep up with the pace of js development so I'll consider any opinions.

**Comment by flosse** [comment]
> I find it hard to keep up with the pace of js development so I'll consider any opinions.

yeah... this is one big reason I left the Node.js world! In rust the compiler tells me **that** there are conflicts and **what** the conflicts are. So there are **no** surprises at run-time. Building an application with Node is easy and takes little effort **but** I don't wont to be responsible for maintenance ;-)
Nevertheless building UIs without web technologies is often hard. So in this case I see two options for a more reliable/relaxed long term way:
1. Decrease dependencies and "manually" implement the functionalities with the main framework (React should be stable for a while now) or
2. Render the app on the server

What do you think?

**Comment by cmsd2** [comment]
in the future we might be able to write isomorphic webapps purely in rust by compiling the rust code to webassembly for the client.

here and now we have to choose between fat frameworks like asp.net and rails that incorporate features like data binding, or api frameworks that leave all the lifting to the javascript. right now it's much easier in rust to do the latter since nobody's written a proper framework yet.

i don't think this particular project is the last word in rust webapp design. it's only meant to get the conversation started.

there's another forms component for react out there, or i could try doing the bits from scratch.
i don't want to end up as the defacto maintainer of a fork of the existing easy forms project -- i don't use angular enough to make that work.
